src/queryBuilder/FilterContents.js

```diff
@@ -42,13 +43,13 @@ let FilterContents = inject(_.defaults)(
             label={nodeField ? nodeLabel : 'Pick a Field'}
             options={fieldsToOptions(fields)}
             // TODO: consider type options in case this isn't safe, e.g. a field/type change action
-            onChange={field => root.mutate(node.path, { field })}
+            onChange={field => tree.mutate(node.path, { field })}
           />
           {nodeField && (
             <div style={{ margin: '0 5px' }}>
               <select
-                onChange={({ target: { value } }) => {
-                  root.typeChange(node, value)
+                onChange={({ target: { value: type } }) => {
+                  tree.replace(node.path, newNodeFromType(type, fields, node))
```

Comment:
With a contexture-client-side `transformReplace` (that would take a node and a transform, and simply call `replace` with them), we could do this here instead:

```js
tree.transformReplace(newNodeFromType(type, fields), node)
```

I find this preferably, personally, but it's really not that much of a difference. Just thought I'd throw it out there.